cfme/base/login.py

```diff
@@ -51,35 +70,31 @@ def current_fullname(self):
 
     @property
     def current_groupname(self):
-        # TODO: Move these locators and accessors to a widget "Group Dropdown" widget
-        # 5.9 Locators for finding current group when user has multiple groups
-        current_group_locator = '{}/ul/li/a[@title="Currently Selected Group"]'.format(
-            self.group_list_locator)
-        current_group_marker = ' (Current Group)'
-        try:
-            current_group = self.browser.element(current_group_locator)
-            return self.browser.text(current_group).replace(current_group_marker, '')
-        except NoSuchElementException:
-            return self.settings.items[1].strip()
+        current_groups = self.settings.groups.items
+
+        # User is only assigned to one group
+        if len(current_groups) == 1:
+            return current_groups[0]
+
+        for group in current_groups:
+            if self.settings.groups.SELECTED_GROUP_MARKER in group:
+                return group.replace(self.settings.groups.SELECTED_GROUP_MARKER, '')
+
+        # Handle some weird case where we don't detect a current group
```

Comment:
**Required**

Let's move this into an `else` block on the for loop, so the behavior here is a bit more explicit and this raise isn't just hanging off the end.

```python
for group in current_groups:
    if check:
        return group.replace.yadayada
else:
    # This block executes when return isn't hit in the for loop
    raise ShitsBroke('no active group')

Reply:
Updated